adcc/ExcitedStates.py

```diff
@@ -667,47 +664,40 @@ class Excitation:
         excitations = []
         for i in range(self.size):
             e = Excitation()
-            e.idx = i
+            e.index = i
             e.__parent_state = self
             for key in dir(self):
                 if key == "excitations":
                     continue
                 if "__" in key or key.startswith("_"):
                     continue  # skip "private" fields
-                if hasattr(type(self), key):
-                    if not isinstance(getattr(type(self), key), property):
-                        continue
-                else:
+                if not hasattr(type(self), key):
+                    continue
+                if not isinstance(getattr(type(self), key), property):
                     continue
 
                 fget = getattr(type(self), key).fget
                 if not hasattr(fget, "__excitation_property"):
-                    # print(dir(fget), key, "not found.")
                     continue
                 # Extract the kwargs passed to mark_excitation_property
                 kwargs = getattr(fget, "__excitation_property").copy()
 
-                def payload(self, key=key, kwargs=kwargs):
-                    return getattr(self.__parent_state, key)[self.idx]
+                def get_parent_property(self, key=key, kwargs=kwargs):
+                    return getattr(self.__parent_state, key)[self.index]
 
-                setattr(type(e), kwargs["name"], property(payload))
+                setattr(type(e), kwargs["name"], property(get_parent_property))
 
                 transform = kwargs.pop("transform_to_ao", False)
                 if transform:
-                    def payload_transform(self, key=key):
-                        return transform_to_ao(
-                            getattr(self.__parent_state, key)[self.idx])
+                    def get_parent_property_transform(self, key=key):
+                        matrix = getattr(self.__parent_state, key)[self.index]
+                        # TODO: make it a one-liner once the lazy tensor stuff
+                        # is merged
```

Comment:
By the way: This just waits for your approval ;)

Reply:
Yes, I know. I think we should merge the lazy stuff first, because refactoring `ExcitedStates` will take some time and thinking (with deprecations, "copy"-ctor, etc.).